R/predict.R

```diff
@@ -72,6 +72,26 @@ predict.WrappedModel = function(object, task, newdata, subset = NULL, ...) {
     newdata = newdata[subset, , drop = FALSE]
   }
 
+
+  # set factor levels, present in test but missing in train, to NA
+  if (model$learner$fix.factors.prediction == TRUE &&
+      any(vcapply(newdata, function(x) class(x)) == "factor") &&
+      any(class(model$learner.model) == "lm" | class(model$learner.model) == "glmmPQL")) {
+    # sometimes we have no task here, e.g. in test_tune_tuneMBO@34
+    if (!missing(task)) {
+      subset = task$env$data[subset, ]
+    } else {
+      subset = newdata
+    }
```

Comment:
> And it looks like before subset was a set of indices and now it's the actual data.

Yeah, confusing. Changed the object name to `newdata` so that `subset` is still the set of indices